File: src/StripController/stripper/SetPixelColorBulkCommand.cpp
#include "SetPixelColorBulkCommand.h"

SetPixelColorBulkCommand::SetPixelColorBulkCommand(Strip* strip, Logger* logger)
	:_strip(strip), _logger(logger)
{
}

int SetPixelColorBulkCommand::GetCommandId()
{
	return Command::SetPixelColorBulkCommandId;
}

void SetPixelColorBulkCommand::Handle(byte* command)
{
	_logger->Trace("Handle SetPixelColorBulkCommand begin");

	int offset = 1;

	byte start = command[offset++];
	byte end = command[offset++];
	byte r = command[offset++];
	byte g = command[offset++];
	byte b = command[offset++];

	char buffer[255];
	sprintf(buffer, "Arguments: {\"start\": %d, \"end\": %d, \"r\": %d, \"g\": %d, \"b\": %d}", start, end, r, g, b);
	_logger->Trace(buffer);

	for(int i = start; i < end; i++)
		_strip->SetPixelColor(i, r, g, b);

	_logger->Trace("Handle SetPixelColorBulkCommand end");
}

File: src/StripController/StripController.Infrastructure/StripWrapper/SetPixelColorBulkCommand.cs
namespace StripController.Infrastructure.StripWrapper
{
    class SetPixelColorBulkCommand : Command
    {
        private readonly byte _start;
        private readonly byte _end;
        private readonly byte _r;
        private readonly byte _g;
        private readonly byte _b;
        
        public SetPixelColorBulkCommand(byte start, byte end, byte r, byte g, byte b)
        {
            _start = start;
            _end = end;
            _r = r;
            _g = g;
            _b = b;
        }

        public override byte[] GetBody()
        {
            return new[] { _start, _end, _r, _g, _b };
        }

        public override int GetCommandId()
        {
            return Commands.SetPixelColorBulkCommandId;
        }
    }
}